vector/src/main/java/im/vector/activity/VectorHomeActivity.java

```diff
@@ -994,8 +972,13 @@ private void updateSelectedFragment(final MenuItem item) {
                     fragment = new CommunicateHomeFragment();
                 }
                 mCurrentFragmentTag = TAG_FRAGMENT_HOME;
-                mSearchView.setVisibility(View.VISIBLE);
-                mSearchView.setQueryHint(getString(R.string.home_filter_placeholder_home));
+                if(getResources().getBoolean(R.bool.enable_riot_search_view)) {
```

Comment:
Please centralise lines 975 to 978, 990 to 993, 1013 to 1016, 1025 to 1028 into a method:
e.g.  
```
private void configureSearchView(String queryHint) {
   if(getResources().getBoolean(R.bool.enable_riot_search_view)) {
        mSearchView.setVisibility(View.VISIBLE);
        mSearchView.setQueryHint(getString(queryHint));
   }
}
```

Reply:
added a function to set the hint